pkg/virt-launcher/virtwrap/converter/compute/memory_test.go

```diff
@@ -0,0 +1,210 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright The KubeVirt Authors.
+ *
+ */
+
+package compute_test
+
+import (
+	"encoding/xml"
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/resource"
+	k8smeta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/converter/compute"
+)
+
+var _ = Describe("Memory Domain Configurator", func() {
+	DescribeTable("should calculate memory in bytes", func(quantity string, bytes int) {
+		vmi := libvmi.New(
+			libvmi.WithMemoryRequest(quantity),
+		)
+
+		var domain api.Domain
+		configurator := compute.MemoryConfigurator{}
+		Expect(configurator.Configure(vmi, &domain)).To(Succeed())
+
+		xmlBytes, err := xml.MarshalIndent(domain.Spec, "", "  ")
+		Expect(err).ToNot(HaveOccurred())
+
+		xml := string(xmlBytes)
+		Expect(strings.Contains(xml, fmt.Sprintf("<memory unit=\"b\">%d</memory>", bytes))).To(BeTrue(), xml)
+	},
+		Entry("specifying memory 64M", "64M", 64*1000*1000),
+		Entry("specifying memory 64Mi", "64Mi", 64*1024*1024),
+		Entry("specifying memory 3G", "3G", 3*1000*1000*1000),
+		Entry("specifying memory 3Gi", "3Gi", 3*1024*1024*1024),
+		Entry("specifying memory 45Gi", "45Gi", 45*1024*1024*1024),
+		Entry("specifying memory 2780Gi", "2780Gi", 2780*1024*1024*1024),
+		Entry("specifying memory 451231 bytes", "451231", 451231),
+	)
+	It("should calculate memory in bytes", func() {
+		By("specyfing negative memory size -45Gi")
+		vmi := libvmi.New(
+			libvmi.WithMemoryRequest("-45Gi"),
+		)
+		var domain api.Domain
+		configurator := compute.MemoryConfigurator{}
+		err := configurator.Configure(vmi, &domain)
+		Expect(err).To(HaveOccurred())
+	})
+
+	Context("with v1.VirtualMachineInstance", func() {
+		It("should handle float memory", func() {
+			vmi := libvmi.New(
+				libvmi.WithMemoryRequest("2222222200m"),
+			)
+			var domain api.Domain
+
+			configurator := compute.MemoryConfigurator{}
+			Expect(configurator.Configure(vmi, &domain)).To(Succeed())
+
+			xmlBytes, err := xml.MarshalIndent(domain.Spec, "", "  ")
+			Expect(err).ToNot(HaveOccurred())
+
+			xml := string(xmlBytes)
+
+			Expect(strings.Contains(xml, `<memory unit="b">2222222</memory>`)).To(BeTrue(), xml)
```

Comment:
for consistency with other configurators tests:
```suggestion
			expectedDomain := api.Domain{Spec: api.DomainSpec{
				Memory: api.Memory{
					Value: uint64(2222222),
				},
			}}
			Expect(domain).To(Equal(expectedDomain))
```